androidApp/src/main/java/com/ajay/seenu/expensetracker/android/presentation/screeens/AnalyticsScreen.kt

```diff
@@ -59,6 +58,12 @@ import java.util.Locale
 @Composable
 fun AnalyticsScreen(navController: NavController, viewModel: AnalyticsViewModel = hiltViewModel()) {
 
+    val context = LocalContext.current
+    LaunchedEffect(Unit) {
+        val filter = FilterPreference.getCurrentFilter(context)
+        viewModel.setFilter(context, filter)
+    }
```

Comment:
As discussed check the occurrence of two launchedEffect.

Reply:
After looking into the code, I find that it has to be done for the first time. As we are retaining the last filter state, we have to get that state from the Shared Preference when the Analytics screen is loaded for the first time. So I have to retain both the launched effect.